sql/wsrep_server_state.cc

```diff
@@ -82,3 +84,24 @@ void Wsrep_server_state::destroy()
     mysql_cond_destroy(&COND_wsrep_server_state);
   }
 }
+
+void Wsrep_server_state::handle_fatal_signal()
+{
+  if (m_instance)
+  {
+    /* Galera background threads are still running and the logging may be
+       relatively verbose in case of networking error. Silence all wsrep
+       logging before shutting down networking to avoid garbling signal
+       handler output. */
+    my_safe_printf_stderr("WSREP: Suppressing further logging\n");
```

Comment:
Are you really sure that the code of `my_safe_printf_stderr` is safe to invoke in a signal handler function? `man 7 signal-safety` on my GNU/Linux system does not list even `printf` or any of the stream oriented functions. A straight `write` to `STDERR_FILENO` might be safe.

Where is an invocation of this function added? I missed it in this pull request.

Reply:
sql/sql_signal_handler.cc first changed file in this pull request. Same function is used in mysys/stacktrace.c.